### PR TITLE
Second Task PR

### PR DESCRIPTION
**Pull request by LiorGnieslawMoveo**

### CODE REVIEW

File: src/app/components/navigation/navigation.component.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<nav>
```
Comment (noyflaysher) [inline src/app/components/navigation/navigation.component.html]: Why its not in header component?

File: src/app/components/navigation/navigation.component.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    <a routerLink="/pokemons">Pokemons</a>
+    <a routerLink="/search-history">Search History</a>
+    <a routerLink="/auth">Log-In</a>
+</nav>
```
Comment (noyflaysher) [inline src/app/components/navigation/navigation.component.html]: I want you to use array and do ngFor

File: src/app/app-routing.module.ts [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
 import { RouterModule, Routes } from '@angular/router';
 import { PokemonsComponent } from './components/pokemons/pokemons.component';
+import { HistoryComponent } from './components/history/history.component';
+import { AuthComponent } from './components/auth/auth.component';
 
 const routes: Routes = [
-  { path: 'pokemons', component: PokemonsComponent}
+  { path: 'pokemons', component: PokemonsComponent},
+  { path: 'search-history', component: HistoryComponent},
+  { path: 'auth', component: AuthComponent},
```
Comment (noyflaysher) [inline src/app/app-routing.module.ts]: change to login

File: src/app/components/pokemons/pokemons.component.ts [hunk 4]
```diff
@@ -41,10 +43,19 @@ export class PokemonsComponent implements OnInit {
           this.getAllTypes();
         });
   }
+
   searchByName(event: string): void {
     this.filteredPokemons = this.pokemons.filter((pokemon: Pokemon) =>
       pokemon.name.toLowerCase().includes(this.searchInput.toLowerCase())
     );
+    
+    if (this.searchInput.length > 0) {
```
Comment (noyflaysher) [inline src/app/components/pokemons/pokemons.component.ts]: why do you need this check?
if the length is 0?

File: src/app/components/pokemons/pokemons.component.ts [hunk 5]
```diff
@@ -41,10 +43,19 @@ export class PokemonsComponent implements OnInit {
...
+      this.searchHistory.push(this.searchInput);
+      if (this.searchHistory.length > 5){
+        this.searchHistory.shift();
+      }
+    localStorage.setItem('searchHistory', JSON.stringify(this.searchHistory));
```
Comment (noyflaysher) [inline src/app/components/pokemons/pokemons.component.ts:57]: if all have the same key its override

File: src/app/components/pokemons/pokemons.component.ts [hunk 6]
```diff
@@ -64,4 +75,13 @@ export class PokemonsComponent implements OnInit {
       );
     }
   }
+
+  loadSearchHistory(): void {
+    if (typeof localStorage !== 'undefined') {
```
Comment (noyflaysher) [inline src/app/components/pokemons/pokemons.component.ts]: why?

File: src/app/components/history/history.component.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+@Component({
+  selector: 'app-history',
+  templateUrl: './history.component.html',
+  styleUrl: './history.component.scss'
+})
+
+export class HistoryComponent implements OnInit{
+  searchHistory: string[] = [];
+
+  constructor() {}
```
Comment (noyflaysher) [inline src/app/components/history/history.component.ts]: Remove the constructor if you din;t use it

File: src/app/components/history/history.component.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+  ngOnInit(): void {
+    this.loadSearchHistory();
+  }
+
+  loadSearchHistory(): void {
```
Comment (noyflaysher) [inline src/app/components/history/history.component.ts]: You have this code in pokemons. Think how to do this function once

File: src/app/components/history/history.component.scss [hunk 9]
```diff
@@ -0,0 +1,27 @@
+@import "../../../styles.scss";
```
Comment (noyflaysher) [inline src/app/components/history/history.component.scss]: Try to remove this import

File: src/app/components/history/history.component.scss [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+.history {
```
Comment (noyflaysher) [inline src/app/components/history/history.component.scss:1]: use nesting

File: src/app/components/header/header.component.scss [hunk 11]
```diff
@@ -13,5 +13,19 @@
...
+  } 
+
+  .logout-button{
+    padding: 8px;
+    font-size: 16px;
+    border: var(--border-style);
+    border-radius: 4px;
+    margin-right: 10px;
+    background-color: var(--red-color);
+    color: var(--pokemon-yellow);
```
Comment (noyflaysher) [inline src/app/components/header/header.component.scss]: change in variables.scss the color name

File: src/app/components/header/header.component.ts [hunk 12]
```diff
@@ -1,10 +1,27 @@
...
 })
-export class HeaderComponent {
-  title = 'Pokemon App';
+export class HeaderComponent implements OnInit {
+  isLoggedIn: boolean = false;
+
+  constructor(private router: Router) {}
+
+  ngOnInit(): void {
+    if (typeof sessionStorage !== 'undefined') {
```
Comment (noyflaysher) [inline src/app/components/header/header.component.ts]: why do you check this?

File: src/app/components/header/header.component.ts [hunk 13]
```diff
@@ -1,10 +1,27 @@
...
+      this.isLoggedIn = sessionStorage.getItem('loggedInEmail') === 'true';
```
Comment (noyflaysher) [inline src/app/components/header/header.component.ts]: why you use in session?

File: src/app/components/header/header.component.ts [hunk 14]
```diff
@@ -1,10 +1,27 @@
...
+    }
```
Comment (noyflaysher) [inline src/app/components/header/header.component.ts]: if the value of something if true you can check
if(sessionStorage.getItem('loggedInEmail')){}

File: src/app/components/header/header.component.ts [hunk 15]
```diff
@@ -1,10 +1,27 @@
...
+  }
+
+  logout(): void {
+    sessionStorage.clear();
```
Comment (noyflaysher) [inline src/app/components/header/header.component.ts]: why you clear all? If you have another things you need ?

File: src/app/components/header/header.component.ts [hunk 16]
```diff
@@ -1,10 +1,27 @@
...
+    this.isLoggedIn = false;
+    this.router.navigate(['/auth']).then(() => {
+      window.location.reload();
```
Comment (noyflaysher) [inline src/app/components/header/header.component.ts]: why do you need reload?

File: src/app/components/auth/auth.component.ts [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+@Component({
+  selector: 'app-auth',
+  templateUrl: './auth.component.html',
+  styleUrl: './auth.component.scss'
+})
+export class AuthComponent implements OnInit {
+  email: string = '';
+  isLoggedIn: boolean = false;
+
+  constructor() { }
```
Comment (noyflaysher) [inline src/app/components/auth/auth.component.ts]: remove if you don't use

File: src/app/components/auth/auth.component.ts [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+
+  ngOnInit(): void {
+    if (typeof sessionStorage !== 'undefined') {
```
Comment (noyflaysher) [inline src/app/components/auth/auth.component.ts]: why?

File: src/app/components/auth/auth.component.ts [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+      const storedEmail = sessionStorage.getItem('loggedInEmail');
```
Comment (noyflaysher) [inline src/app/components/auth/auth.component.ts]: changed loggedInEmail to meaningfull name

File: src/app/components/auth/auth.component.ts [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+      if (storedEmail === 'demo@skills.co.il') {
```
Comment (noyflaysher) [inline src/app/components/auth/auth.component.ts]: same 'demo....' in a var

File: src/app/components/auth/auth.component.ts [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+  templateUrl: './auth.component.html',
+  styleUrl: './auth.component.scss'
+})
+export class AuthComponent implements OnInit {
+  email: string = '';
+  isLoggedIn: boolean = false;
+
+  constructor() { }
+
+  ngOnInit(): void {
```
Comment (noyflaysher) [inline src/app/components/auth/auth.component.ts]: you don't need to do this check. if you logged in you save in the storage.

File: src/app/components/header/header.component.ts (hunk 12 above)
Comment (LiorGnieslawMoveo) [reply src/app/components/header/header.component.ts]: there is an error if the session storage is not initialized yet

File: src/app/components/pokemons/pokemons.component.ts (hunk 6 above)
Comment (LiorGnieslawMoveo) [reply src/app/components/pokemons/pokemons.component.ts]: there is an error if the local storage is not initialized yet

File: src/app/components/header/header.component.ts [hunk 22]
```diff
@@ -1,10 +1,38 @@
...
   templateUrl: './header.component.html',
   styleUrl: './header.component.scss'
 })
-export class HeaderComponent {
-  title = 'Pokemon App';
-}
+export class HeaderComponent implements OnInit {
+  isLoggedIn: boolean = false;
+
+  navLinks: any[] = [
```
Comment (noyflaysher) [inline src/app/components/header/header.component.ts]: Please extract it to constants folder -> navLinks.ts

File: src/app/components/history/history.component.html [hunk 23]
```diff
@@ -0,0 +1,9 @@
+<div class="history" *ngIf="searchHistory.length > 0">
```
Comment (noyflaysher) [inline src/app/components/history/history.component.html]: use if else

File: src/app/components/header/header.component.ts [hunk 24]
```diff
@@ -1,10 +1,38 @@
...
+
+  constructor(
+    private pokemonService: PokemonService,
+    private router: Router,
+    private cookieService: CookieService
+    ) {}
+
+  ngOnInit(): void {
+    this.isLoggedIn = this.cookieService.get('isLoggedIn') === 'true';
+    this.pokemonService.getIsLoggedInSubject().subscribe(isLoggedIn => {
```
Comment (noyflaysher) [inline src/app/components/header/header.component.ts]: I don't understand why you use -  this.pokemonService.getIsLoggedInSubject()

File: src/app/components/history/history.component.ts [hunk 25]
```diff
@@ -0,0 +1,42 @@
...
+
+  constructor(
+    private pokemonService: PokemonService,
+    private router: Router,
+    private cookieService: CookieService
+    ) {}
+
+  ngOnInit(): void {
+    this.isLoggedIn = this.cookieService.get('isLoggedIn') === 'true';
+    this.pokemonService.getIsLoggedInSubjectStatus().subscribe(isLoggedIn => {
```
Comment (noyflaysher) [inline src/app/components/history/history.component.ts:27]: vYou dont need this

File: src/app/components/login/login.component.ts [hunk 26]
```diff
@@ -0,0 +1,42 @@
...
+
+  constructor(
+    private pokemonService: PokemonService,
+    private router: Router,
+    private cookieService: CookieService
+    ){}
+
+  ngOnInit(): void {
+    this.isLoggedIn = this.cookieService.get('isLoggedIn') === 'true';
+    this.pokemonService.getIsLoggedInSubjectStatus().subscribe(isLoggedIn => {
```
Comment (noyflaysher) [inline src/app/components/login/login.component.ts:26]: dont need it

File: src/app/components/login/login.component.ts [hunk 27]
```diff
@@ -0,0 +1,42 @@
...
+      this.isLoggedIn = isLoggedIn;
+    });
+
+  }
+
+  login(): void {
+    if (this.email === 'demo@skills.co.il') {
```
Comment (noyflaysher) [inline src/app/components/login/login.component.ts]: move it to var

File: src/app/components/login/login.component.ts [hunk 28]
```diff
@@ -0,0 +1,42 @@
...
+      this.cookieService.set('isLoggedIn', 'true');
+      this.isLoggedIn = true;
+      this.pokemonService.setLoggedInStatus(true);
+      this.router.navigate(['/pokemons']);
+      console.log('logged in to pokemons')
```
Comment (noyflaysher) [inline src/app/components/login/login.component.ts]: remove

File: src/app/components/history/history.component.ts [hunk 29]
```diff
@@ -0,0 +1,42 @@
...
+      this.isLoggedIn = isLoggedIn;
+    });
+    if (!this.isLoggedIn){
+      this.router.navigate(['/login']);
+    }
+    this.loadSearchHistory();
+  }
+
+  loadSearchHistory(): void {
+    if (typeof localStorage !== 'undefined') {
```
Comment (noyflaysher) [inline src/app/components/history/history.component.ts]: you have this too on Pokemons component. Write it just one time

File: src/app/services/pokemon.service.ts [hunk 30]
```diff
@@ -1,18 +1,26 @@
...
-import { Pokemon } from '../type/pokemon';
+import { Pokemon } from '../interfaces/pokemon.interface';
+import { CookieService } from 'ngx-cookie-service';
 
 @Injectable({
   providedIn: 'root'
 })
 export class PokemonService {
   private apiUrl = 'https://pokeapi.co/api/v2/pokemon/';
+  private isLoggedInSubject: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(this.cookieService.get('isLoggedIn')==='true');
```
Comment (noyflaysher) [inline src/app/services/pokemon.service.ts:14]: i think its Unnecessary. Try to remove it and the set and get function

**Pull request merged**